src/SplitProvider.tsx

```diff
@@ -1,30 +1,22 @@
 import { SplitFactory } from '@splitsoftware/splitio';
-import React, { createContext } from 'react';
+import React, { createContext, useEffect, useState } from 'react';
 
-import {
-  ISplitContextValues,
-  ISplitProviderProps,
-  SplitReactContext,
-} from './types';
+import { ISplitContextValues, ISplitProviderProps } from './types';
 
 /**
  * Creating a React.Context with default values.
- * @returns {SplitReactContext}
  */
-export const SplitContext: SplitReactContext = createContext<
-  ISplitContextValues
->({
-  client: {} as SplitIO.IClient,
+export const SplitContext = createContext<ISplitContextValues>({
+  client: null,
```

Comment:
I realized I kept both your `isReady` and my client nullable. I don't think both of these are necessary. I've already given my reasons for making client nullable (and still hiding that fact in our consumer components/hooks). And `isReady` can be the same as `client != null`. Thoughts?

Reply:
`isReady` is a SDK related attribute not necessarily whether `client` is null or not

Reply:
True....don't think we could hide this though?

Reply:
We could, wouldn't be good to share when the SDK is ready thou?

Reply:
I'm not sure. The docs say that all split values return `control` as the default value in the case that something fails or is not ready. My hook does the same thing (I'm not sure if the `Split` component got updated with that though).
If the implementor is trying to exhaustively handle all cases it seems that "not ready" is the same as `"control"` value.
https://help.split.io/hc/en-us/articles/360020528072-Control-treatment

If you don't agree I'll back down. It would only matter to people if they are using the context directly instead of our hooks / HOC.

Reply:
Is this a new feature in the SDK? Because before when the SDK was not ready we used to get a `treatment: null` when SDK was not ready. But even returning a default `control` treatment, users might not know what that is, the `isReady` would be a simple (maybe dumb) way to know that.

Reply:
I'm not sure if it is a new feature. I just saw it in the docs. I'm not sure it is this libraries job to inform users how Split works. We are bound by their constraints after all.

I'm fine keeping it. At this point I'd rather give implementors more information than less. This `isReady` can easily be hidden by a hook or HOC. And it even is in my hooks, but we can talk about that in that PR not here.

Reply:
Sounds good to me.